src/parsers/espresso/parser.js

```diff
@@ -0,0 +1,255 @@
+        return units;
+    }
+
+    getName() {
+        return this.data.control.title;
+    }
+
+    /**
+     * @summary Return unit cell parameters from CELL_PARAMETERS card
+     * @param {String} text - cards data
+     * @return {{cell: Number[][], units: String}}
+     */
+    getCellConfig(text) {
+        let cell = {};
+        if (this.data.system === undefined)
+            throw new Error("No &SYSTEM section found in input this.data.");
+        if (this.data.system.ibrav === undefined) throw new Error("ibrav is required in &SYSTEM.");
+
+        if (this.data.system.ibrav === 0) {
+            const match = regex.cellParameters.exec(text);
+            if (match) {
+                const units = match[1];
+                const values = match.slice(2, 11);
+                // creating matrix 3 by 3 of numbers from 9 strings
+                const vectors = Array.from({ length: 3 }, (_, i) =>
+                    values.slice(i * 3, i * 3 + 3).map(Number),
+                );
+                cell = { cell: vectors, units };
+                cell.type = "TRI"; // TODO: implement type detection, now defaults to TRI
+                return cell;
+            }
+        } else {
+            cell = this.ibravToCellConfig(this.data.system);
+            return cell;
+        }
+        throw new Error("Couldn't read cell parameters");
+    }
+
+    /**
+     * @summary Returns cell config from ibrav and celldm(i) parameters
+     *
+     * QE docs: https://www.quantum-espresso.org/Doc/INPUT_PW.html#ibrav
+     * "If ibrav /= 0, specify EITHER [ celldm(1)-celldm(6) ]
+     *   OR [ A, B, C, cosAB, cosAC, cosBC ]
+     *   but NOT both. The lattice parameter "alat" is set to
+     *   alat = celldm(1) (in a.u.) or alat = A (in Angstrom);"
+     *
+     * @param {Object} system - The system parameters from &SYSTEM namelist
+     * @param {Number} system.ibrav - ibrav parameter
+     * @param {Number[]} [system.celldm] - celldm parameters
+     * @param {Number} [system.a] - A parameter in angstroms
+     * @param {Number} [system.b] - B parameter in angstroms
+     * @param {Number} [system.c] - C parameter in angstroms
+     * @param {Number} [system.cosab] - cosAB parameter
+     * @param {Number} [system.cosac] - cosAC parameter
+     * @param {Number} [system.cosbc] - cosBC parameter
+     * @returns {{cell: Number[][], type: String}}
+     */
+    ibravToCellConfig(system) {
+        const { ibrav, celldm, a, b, c, cosab, cosac, cosbc } = system;
+        if (celldm && a) {
+            throw new Error("Both celldm and A are given");
+        } else if (!celldm && !a) {
+            throw new Error("Missing celldm(1)");
+        }
+
+        const type = this.ibravToCellType(ibrav);
+        const [_a, _b, _c] = this.getLatticeConstants(celldm, a, b, c);
+        const [alpha, beta, gamma] = this.getLatticeAngles(celldm, cosbc, cosac, cosab);
+
+        const lattice = new Lattice({
+            a: _a,
```

Comment:
This thing with `_a`, `_b`, `_c`, `_units` just seems odd - why not use just a, b, c and `let` instead of `const`?